lib/cn/csched_sp3.c

```diff
@@ -1661,14 +1660,14 @@ sp3_submit(struct sp3 *sp, struct cn_compaction_work *w, uint qnum)
             break;
     }
 
-    /* Force compaction reads to use mcache if the value blocks for this node reside on
+    /* Force compaction reads to use memory maps if the value blocks for this node reside on
      * the pmem media class. This is not accurate if the mclass policy is changed during
      * subsequent kvs opens, which results in a mix of media classes for the k/vblocks
      * in this node. However, this is not a correctness issue and will recover on its own
      * after a series of compaction operations.
      */
     if (cn_tree_node_mclass(tn, HSE_MPOLICY_DTYPE_VALUE) == HSE_MCLASS_PMEM) {
-        w->cw_iter_flags |= kvset_iter_flag_mcache;
+        w->cw_iter_flags |= kvset_iter_flag_mmap;
```

Comment:
We should probably only do this for the root node.  @nabeelmmd ???

Reply:
Make a mental note or a ticket and do that in a follow on PR.  I want this PR to have no functional changes.

Reply:
Greg, if you are worried about the page cache pressure in using mmap() for compaction reads on all the nodes, mmap() on persistent memory does not involve the page cache, i.e., there's no page fault. The memory accesses are purely via load/store completely bypassing the kernel.

Reply:
Wow, that's cool!  Only have to worry about 128T drives consuming all the virtual address space then ;-)
I think newer CPUs have 52-bits of virtual addresses, but I don't think I've actually seen one yet...